### PR TITLE
Docs: Add Windows notes, .env section, make formatting consistent, fix typos to CONTRIBUTING.md

### LINKED ISSUE #462: Add `npm run postinstall` to contributing.md
### Describe the bug:

The _.env_ file is mandatory for the project to build, however it is intentionally not provided as part of the project as it will contain the project secrets. The Chapter developer/installer is required to issue the proper commands to create and edit the _.env_ file.

A script has been provided to copy defaults values from the _.env.example_ into the _.env_ .

Put the following sentence into the CONTRIBUTING.md file around line 310 in
the section run the app step 1.

> Run `npm run postinstall` to copy the file _.env.example_ into the file _.env_.
> If the _.env_ file exists it will not be overwritten.

### Tell us about your browser and operating system:
- Browser(s) name and version:  __terminal__
- Operating System: 
uname -ior
5.3.7-301.fc31.x86_64 x86_64 GNU/Linux

### LINKED ISSUE #465: Handling symlinks on Windows
@paizle 's issue seems to be related to Git Bash + Windows and probably because there's a symlink involved.

image

@ojeytonwilliams do we think using the literal path instead of a symlink is likely to fix one environment and break others? If so, perhaps we break that into a new issue.

I found some chat about the [escaping of characters being an issue / bug with Git Bash on Windows](https://github.com/freeCodeCamp/chapter/issues/447#issuecomment-794076923) so symlink and/or escaping seems like they could the what's breaking Git Bash.

_Originally posted by @allella in https://github.com/freeCodeCamp/chapter/issues/461#issuecomment-809370861_

### PR DESCRIPTION
**Pull request by allella**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `master` branch of Chapter.

* Adds notes and improved the docs for Windows / Git for Windows / Git Bash users to start addressing #465 @paizle
* Added a **.env Configuration File** section and link to it to avoid confusion with .env and .env.example like in #462 @NorthDecoder 
* Made commands, command line output, and italics text consistent in the various sections
* Fix typos noted in #447 
* Format "Note:" text consistently
* Made "Prerequisite" software bold and consistent.
* Added better OS-specific notes and links for downloading Git, Docker, and Node.js

Edit(Oliver): 
Closes #465 
Closes #462

I hope that edit is okay, @allella.  I wasn't sure it closed 447, so I left that alone.

### CODE REVIEW

**Comment by allella** [comment]
@paizle and @NorthDecoder I wanted to add you add you as reviewers but GitHub won't let me. I think we figured it's because you haven't already commented on the PR and/or something to do with starring the project or the freeCodeCamp team settings.

In any case, I'd appreciate if you can look at the changes.  Viewing it at https://github.com/allella/chapter/blob/docs/windows-contributing/CONTRIBUTING.md is the only sane way to review the changes.

**Comment by paizle** [comment]
> 
> 
> @paizle and @NorthDecoder I wanted to add you add you as reviewers but GitHub won't let me. I think we figured it's because you haven't already commented on the PR and/or something to do with starring the project or the freeCodeCamp team settings.
> 
> In any case, I'd appreciate if you can look at the changes. Viewing it at https://github.com/allella/chapter/blob/docs/windows-contributing/CONTRIBUTING.md is the only sane way to review the changes.

The docs look ok to me. As far as I can tell, Git Bash on Windows seems to work fine for me at this point.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,86 +43,76 @@ Follow these steps to fork the repository:
...
-    ```
+    If you are not on master or your working directory is not clean, resolve any outstanding files/commits and checkout _master_:
 
-    If you are not on master or your working directory is not clean, resolve any outstanding files/commits and checkout `master`:
     ```sh
     git checkout master
     ```
 
-2. Sync the latest changes from the chapter upstream `master` branch to your local master branch. This is very important to avoid conflicts later.
+2. Sync the latest changes from the upstream **Chapter** _master_ branch to your local fork's _master_ branch. This is very important to keep things syncronzied and avoid "merge conflicts".
```
Comment (ojeytonwilliams) [inline CONTRIBUTING.md:113]:
```suggestion
2. Sync the latest changes from the upstream **Chapter** _master_ branch to your local fork's _master_ branch. This is very important to keep things synchronized and avoid "merge conflicts".
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
This looks good  All I found was a single typo.

**Comment by allella** [comment]
@nhcarrigan could you take a look at **Contributing Code >> Step 2 - Prepare the Development Environment** of my PR for [CONTRIBITING.md](https://github.com/allella/chapter/blob/docs/windows-contributing/CONTRIBUTING.md) to see if the Windows related terminal / shell notes look good?

**Comment by nhcarrigan** [comment]
That section looks good to me. 🙂

**Comment by allella** [comment]
@all-contributors add @nhcarrigan @ojeytonwilliams for docs

**Pull request merged**